File: python/test/test_17.py
from python.src.code_17 import Solution
from python.test.python_deps.boilerplate import BoilerTest


def input_fetcher(case):
    return case["input"]["digits"]


def test():
    test = BoilerTest(17, input_fetcher)
    solution = Solution().letterCombinations
    test.run_tests(solution)

File: cpp/test/test_17.cpp
#include <algorithm>

#include "../src/code_17.cpp"
#include "cpp_deps/boilerplate.hpp"

void test(Solution& sol, const json& input, const json& output) {
    string digits = input["digits"].get<string>();
    vector<string> expected = output.get<vector<string>>();
    vector<string> result = sol.letterCombinations(digits);

    sort(result.begin(), result.end());

    CHECK_EQ(result, expected);
}

// asdf
TEST_CASE("") {
    TEST(17);
}